src/test/java/ActionableMessageHelperImplTest.java

```diff
@@ -0,0 +1,138 @@
+  private static final String TO_HEADER_VALUE_TOO_MANY_RECIPIENTS =
+      createHeaderWithNRecipients(
+          ActionableMessageHelperImpl.MAILING_LIST_LOWER_BOUND, TO_HEADER_VALUE_USER_EMAIL, ", ");
+  private static final MessagePartHeader toHeaderUserEmail =
+      new MessagePartHeader().setName("To").setValue(TO_HEADER_VALUE_USER_EMAIL);
+  private static final MessagePartHeader toHeaderNotUserEmail =
+      new MessagePartHeader().setName("To").setValue(TO_HEADER_VALUE_NOT_USER_EMAIL);
+  private static final MessagePartHeader toHeaderUndisclosedRecipients =
+      new MessagePartHeader().setName("To").setValue(TO_HEADER_VALUE_UNDISCLOSED_RECIPIENTS);
+  private static final MessagePartHeader toHeaderTooManyRecipients =
+      new MessagePartHeader().setName("To").setValue(TO_HEADER_VALUE_TOO_MANY_RECIPIENTS);
+
+  private static final Message starredMessage =
+      new Message()
+          .setPayload(new MessagePart().setHeaders(Collections.singletonList(toHeaderNotUserEmail)))
+          .setLabelIds(LABELS_STARRED);
+  private static final Message importantMessage =
+      new Message()
+          .setPayload(new MessagePart().setHeaders(Collections.singletonList(toHeaderNotUserEmail)))
+          .setLabelIds(LABELS_IMPORTANT);
+  private static final Message undisclosedRecipientsMessage =
+      new Message()
+          .setPayload(
+              new MessagePart()
+                  .setHeaders(Collections.singletonList(toHeaderUndisclosedRecipients)));
+  private static final Message tooManyRecipientsMessage =
+      new Message()
+          .setPayload(
+              new MessagePart().setHeaders(Collections.singletonList(toHeaderTooManyRecipients)));
+  private static final Message notUserEmailMessage =
+      new Message()
+          .setPayload(
+              new MessagePart().setHeaders(Collections.singletonList(toHeaderNotUserEmail)));
+  private static final Message userEmailMessage =
+      new Message()
+          .setPayload(new MessagePart().setHeaders(Collections.singletonList(toHeaderUserEmail)));
+  private static final Message messageWithoutToHeader = new Message();
+
+  private static String createHeaderWithNRecipients(int n, String headerValue, String delimiter) {
+    StringBuilder stringBuilder = new StringBuilder();
+    for (int i = 0; i < n; i++) {
+      stringBuilder.append(headerValue).append(delimiter);
+    }
+
+    return stringBuilder.toString();
+  }
+
+  @Test(expected = GmailMessageFormatException.class)
+  public void getPriorityNoHeaders() {
+    actionableMessageHelperImpl.assignMessagePriority(messageWithoutToHeader, USER_EMAIL);
+  }
+
+  @Test
+  public void getPriorityStarredMessage() {
+    ActionableMessage.MessagePriority actualPriority =
+        actionableMessageHelperImpl.assignMessagePriority(starredMessage, USER_EMAIL);
+    Assert.assertEquals(ActionableMessage.MessagePriority.HIGH, actualPriority);
+  }
+
+  @Test
+  public void getPriorityImportantMessage() {
```

Comment:
What about important and starred? Other combinations to check precedence?

Reply:
Fixed! Thanks for the suggestion